### ISSUE TITLE
[SF Migration] Introduce unit tests for BO theme

### ISSUE THREAD

**Issue by matks** [PrestaShop/PrestaShop#13423]
We need to introduce a setup for unit tests for the javascript code written for the new BO theme.

@PierreRambaud I think you have experience on this ?

**Comment by PierreRambaud** [comment]
Yes, I did it for the facetedsearch module, but I don't have many thing in this repository (https://github.com/PrestaShop/ps_facetedsearch).
Which means I don't try to create mocks, stubs and anything complicated.

But I can recommend:
- mocha  (https://github.com/mochajs/mocha)
- Chai for assertions: https://www.chaijs.com/ (can play with dom (chai-dom) and have many plugins to not rewrite the wheel)
- Sinon.js for mocks and stubs (https://sinonjs.org/)

**Comment by sarjon** [comment]
Is there a decision made of what tools/approach we use?

**Comment by PierreRambaud** [comment]
Not really, just consider to use https://github.com/PrestaShop/eslint-config as lint configuration.
